Dynatrace/dynatrace-operator
# Redeploy ActiveGate incase the KSPM token changes

`SetHash` is an inferior version of `hasher.AddAnnotation` as it only works for StatefulSets and is not nil safe

## Description

In the unfortunate case when the `<dk>-kspm-token` secret's content changes, we want to redeploy the ActiveGate, to make sure it doesn't get out of sync with components that will also us `<dk>-kspm-token`.
- As the `<dk>-kspm-token` is only create once (or when it's deleted by the user), this is just a failsafe.

See commits/commit-description for more details

## How can this be tested?

1. Deploy kspm:
```
apiVersion: dynatrace.com/v1beta3
kind: DynaKube
metadata:
  name: dk
  namespace: dynatrace
spec:
  apiUrl: https://<tenant>.dev.dynatracelabs.com/api
  kspm:
    enabled: true
  activeGate:
    capabilities:
      - kubernetes-monitoring
```
2. Delete `<dk>-kspm-token` Secret
3. See that the ActiveGate gets restarted.
